core/src/main/java/org/ehcache/core/spi/time/TickingTimeSource.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.core.spi.time;
+
+import org.ehcache.spi.service.Service;
+import org.ehcache.spi.service.ServiceProvider;
+
+import java.util.Timer;
+import java.util.TimerTask;
+
+/**
+ * The default {@link TimeSource} in Ehcache. it increases the time every millisecond
+ * using a timer. We set it to the system time every second to prevent time drifting.
+ */
+public class TickingTimeSource implements TimeSource, Service {
+
+  private static final long GRANULARITY_IN_MS = 1L;
```

Comment:
I'm thinking there must be numerous cases where 100ms would be fine for a granularity... why such a short default?  If we make this require configuration then we could tie it to the other time related configuration parameters of the cache.

Reply:
You mean no default?

Reply:
I mean *if* this becomes a non-default then it might be best to not have a default and force the user to specify. If it becomes the default then thats a different question.

Reply:
Ok. That's what I did